### DISCUSSION TITLE
怎么给群内的成员上头衔? 怎么监听到机器人自己@自己的消息?

### DISCUSSION THREAD

**Discussion by onlyLive-123** [simple-robot/Discussions#179] [Q&A] [answered]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

在Mirai里面找到这个https://github.com/MrXiaoM/SpecialTitleYouWant,以插件的形式,是支持上头衔的.

但是在simbot-component-mirai-core 该怎么实现上头衔呢,求指导

有时候想禁言群内某个人,但是又没小号在里面,该怎么监听到机器人发给自己的消息 ,求指导

**Comment by ForteScarlet** [comment, marked as answer]
1. 获取与设置头衔可参考 #53 
2. 监听 at 消息可参考 #176 ，而只监听bot自己被at则可以简化，参考 [76#discussioncomment-4973262](https://github.com/orgs/simple-robot/discussions/76#discussioncomment-4973262)

**Reply by onlyLive-123** [reply]
非常感谢,总体来说功能已经实现
特别需要注意的 做atBot监听的时候一定要添加 `@ContentTrim`注解

还有另外一个问题,就是机器人能在群内 自己@自己 的消息能监听得到吗
为什么问这个? 
就是群主是机器人 想在这个群内给某个人上头衔或者禁言 如果让机器人自己@自己能监听到消息的话 那么实现起来就很容易了

目前经过测试 在群内机器人自己@机器人自己 是监听不到这个消息的

望指教!

**Reply by ForteScarlet** [reply]
在mirai中，在其它账号发送的群消息的事件是 [GroupMessageSyncEvent](https://kdoc.mirai.mamoe.net/2.15.0/mirai-core-api/net.mamoe.mirai.event.events/-group-message-sync-event/index.html) 。此事件类型在 mirai 组件中没有针对性的包装，因此可以监听 [UnsupportedMiraiEvent](https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai.event/-unsupported-mirai-event/index.html) 并判断 `event.originalEvent` 是否为 `GroupMessageSyncEvent` 来实现。缺点是收到的消息可能不能直接转化为 simbot 的 `MessageContent` ，而是mirai自己的消息类型 `MessageChain` 。
可以选择直接使用mirai的消息类型或通过 [MiraiMessageChainContent](https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai.message/-mirai-message-chain-content/index.html) 等手段进行转化。

**Reply by onlyLive-123** [reply]
非常感谢解答 已实现!
